lobby-db/sample_data.sql

```diff
@@ -6,11 +6,20 @@ delete
 from lobby_user;
 delete
 from access_log;
+delete
+from user_role;
+
+insert into user_role(id, name)
+values (1, 'ADMIN'), -- user can add/remove admins and add/remove moderators, has boot/ban privileges
+       (2, 'MODERATOR'), -- user has boot/ban privileges
+       (3, 'PLAYER'), -- standard registered user
+       (4, 'ANONYMOUS'), -- users that are not registered, they do not have an entry in lobby_user table
+       (5, 'HOST'); -- AKA LobbyWatcher, special connection for hosts to send game updates to lobby
```

Comment:
Hmmm given the check I commented on earlier, would it work implementing something like "permission levels", so we'd just have to compare an int for check for permission, or is the permission model too diverse?

Reply:
That was getting tempting. It's important to note that `host` is a singleton permission, only `host` is authorized for the `host` role. That breaks the integer based check with a 'ranking'. Second, for authorization we should lean to having white-lists as much as possible, be explicit so we can avoid accidental new roles coming in.

It might make more sense to have a many to one relationship for users to user roles. I've been debating that a bit. In that case an admin would have 4 roles associated and we could do direct matching. EG:

```
private boolean isAuthorized(String targetRole, List<String> userRoles) {
  return userRoles.contains(targetRole); 
}
```

The thing that is stopping me from doing that is for now things are still pretty simple, but I'm concerned a bit about the data insertion logic. Going that route, we'd probably want to define a 'role-group', which would map a 'group' to many roles, and we could use that as a join table to translate 'user' to a list of roles through their role group.

It's a decision of whether we want the role mappings to be defined in DB or in code. I'm kinda liking going with a 'role-group', having it all be defined in DB, but at this point the value to switch seems kinda low, probably better to avoid the code-churn and revisit if we have more work with roles.